### DISCUSSION TITLE
Does OhMyBash change/flush my `.bash_history` by default?

### DISCUSSION THREAD

**Discussion by mehdiMj-ir** [ohmybash/oh-my-bash#570] [Q&A] [answered]
I look at `lib/history.sh` file and it doesn't remove my history file aka `.bash_history`, even I add
```.bashrc
export HISTSIZE=-1;
export HISTFILESIZE=-1
```
before `source "$OSH"/oh-my-bash.sh` to insure my history is limitless.
what should I do?

**Comment by akinomyoga** [comment, marked as answer]
> Does OhMyBash change/flush my `.bash_history` by default?

No. By default, Oh My Bash configures the unlimited history size by setting empty strings to `HISTSIZE` and `HISTFILESIZE`.

> what should I do?

What would you like to achieve?

- If you don't want to flush `~/.bash_history`, you don't have to do anything. Oh My Bash sets empty strings to `HISTSIZE` and `HISTFILESIZE` so the history will not be truncated.
  Or, even if you don't touch `HISTSIZE` and `HISTFILESIZE` at all, is your `~/.bash_history` truncated? In such a case, what is the result of the command `declare -p HISTSIZE HISTFILESIZE` in an interactive session?
- If you want to truncate `~/.bash_history`, you need to set `HISTSIZE` and `HISTFILESIZE` to the *non-negative* integer value (that represents the maximum number of history entries you want to keep) *after* `source "$OSH"/oh-my-bash.sh`. If you just want to clear everything, you need to set `0` to both the shell variables.
